packages/data/src/components/use-dispatch/use-dispatch-with-map.js

```diff
@@ -41,18 +36,25 @@ const useDispatchWithMap = ( dispatchMap, deps ) => {
 			registry.dispatch,
 			registry
 		);
-		return mapValues( currentDispatchProps, ( dispatcher, propName ) => {
-			if ( typeof dispatcher !== 'function' ) {
-				// eslint-disable-next-line no-console
-				console.warn(
-					`Property ${ propName } returned from dispatchMap in useDispatchWithMap must be a function.`
-				);
-			}
-			return ( ...args ) =>
-				currentDispatchMap
-					.current( registry.dispatch, registry )
-					[ propName ]( ...args );
-		} );
+		return Object.fromEntries(
+			Object.entries( currentDispatchProps ).map(
```

Comment:
`currentDispatchProps` is always expected to be an object as per [`react-redux`'s syntax](https://react-redux.js.org/using-react-redux/connect-mapdispatch#return).